packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/FunctionEditorState.ts

```diff
@@ -255,7 +255,9 @@ export class FunctionEditorState extends ElementEditorState {
   executionResult?: ExecutionResult | undefined; // NOTE: stored as lossless JSON string
   executionPlanState: ExecutionPlanState;
   parametersState: FunctionParametersState;
-  funcRunPromise: Promise<ExecutionResult> | undefined = undefined;
+  funcRunPromise:
+    | Promise<{ executionResult: ExecutionResult; executionTraceID?: string }>
```

Comment:
Nit: it looks like in most places we format it as `Id`, so I think this would be better as `executionTraceId`